Solutions/Ais.Net/Ais/Net/NmeaTagBlockParser.cs

```diff
@@ -76,10 +81,46 @@ public NmeaTagBlockParser(ReadOnlySpan<byte> source)
                     case 'n':
                     case 'r':
                     case 't':
-                        throw new NotSupportedException("Unsupported field type: " + fieldType);
+                        if (throwWhenTagBlockContainsUnknownFields)
+                        {
```

Comment:
Do we need to test both of these cases?

Reply:
Yes, good catch. I've pushed a new commit for this.